priam/src/main/java/com/netflix/priam/tuner/JVMOptionsTuner.java

```diff
@@ -0,0 +1,152 @@
+            configuredJVMOptions.stream().forEach(line -> buffer.append(line).append("\n"));
+            logger.info("Updating jvm.options with following values: " + buffer.toString());
+        }
+
+        //Verify we can write to output file and it is not directory.
+        File file = new File(outputFile);
+        if (file.exists() && !file.canWrite()) {
+            throw new Exception("Not enough permissions to write to file: " + outputFile);
+        }
+
+        //Write jvm.options back to override defaults.
+        Files.write(new File(outputFile).toPath(), configuredJVMOptions);
+    }
+
+    /**
+     * Update the JVM options file for cassandra by updating/removing JVM options {@link IConfiguration#getJVMExcludeSet()} && {@link IConfiguration#getJVMUpsertSet()}, configuring GC {@link IConfiguration#getGCType()}etc.
+     *
+     * @return List of Configuration as String after reading the configuration from jvm.options
+     * @throws Exception when encountered with invalid configured GC type. {@link IConfiguration#getGCType()}
+     */
+    protected List<String> updateJVMOptions() throws Exception {
+        File jvmOptionsFile = new File(config.getJVMOptionsFileLocation());
+        validate(jvmOptionsFile);
+        final GCType configuredGC = config.getGCType();
+
+        //Read file line by line.
+        final Map<String, JVMOption> excludeSet = config.getJVMExcludeSet();
+
+        //Make a copy of upsertSet, so we can delete the entries as we process them.
+        Map<String, JVMOption> upsertSet = config.getJVMUpsertSet();
+
+        //Don't use streams for processing as upsertSet jvm options needs to be removed if we find them
+        //already in jvm.options file.
+        List<String> optionsFromFile = Files.lines(jvmOptionsFile.toPath()).collect(Collectors.toList());
+        List<String> configuredOptions = new LinkedList<>();
+        for (String line: optionsFromFile)
+        {
+            configuredOptions.add(updateConfigurationValue(line, configuredGC, upsertSet, excludeSet));
+        }
+
+        //Add all the upserts(inserts only left) from config.
+        if (upsertSet != null && !upsertSet.isEmpty()) {
+
+            configuredOptions.add("#################");
+            configuredOptions.add("# USER PROVIDED CUSTOM JVM CONFIGURATIONS #");
+            configuredOptions.add("#################");
+
+            configuredOptions.addAll(upsertSet.values().stream()
+                    .map(jvmOption -> jvmOption.toString()).collect(Collectors.toList()));
+        }
+
+        return configuredOptions;
+    }
+
+    private String updateConfigurationValue(final String line, GCType configuredGC, Map<String, JVMOption> upsertSet, Map<String, JVMOption> excludeSet) {
+
+        JVMOption option = JVMOption.parse(line);
+        if (option == null)
+            return line;
+
+        //Is parameter for GC.
+        GCType gcType = GCTuner.isOptionAvailable(option);
```

Comment:
Instead of `isOptionAvailable`,  `getGcType` makes more sense

Reply:
+1